report/Report.cpp

```diff
@@ -45,6 +48,55 @@ Constant *MakeGlobalString(Module *M, std::string str) {
   return ConstantExpr::getBitCast(v, Type::getInt8Ty(Ctx)->getPointerTo());
 }
 
+bool isStructPtrTy(Type *T) {
+  return T->isPointerTy() && T->getPointerElementType()->isStructTy();
+}
+
+/**
+ * @brief Expand a struct value into a vector of its elements
+ * @param v: struct value that can be casted to ConstantStruct
+ * @param expend_level: max number of levels of struct to expand if nested
+ * @return vector of values in the struct
+ */
+std::vector<Value *> ExpandStruct(Value *v, Function *F, int expend_level = 5) {
+  std::vector<Value *> Expanded;
+  if (expend_level == 0) {
+    return Expanded;
+  }
+
+  LLVMContext &Ctx = F->getContext();
+  Type *StructTy = v->getType()->getPointerElementType();
+  auto &entry = F->getEntryBlock();
+  int len = StructTy->getStructNumElements();
+  for (int i = 0; i < len; i++) {
+    // recursively expanding, so the first index is always 0
+    Value *indices[] = {ConstantInt::get(Type::getInt32Ty(Ctx), 0),
+                        ConstantInt::get(Type::getInt32Ty(Ctx), i)};
+    auto *gep = GetElementPtrInst::CreateInBounds(StructTy, v, indices, "",
+                                                  &*entry.begin());
+    Type *BaseTy = gep->getType()->getPointerElementType();
+    if (BaseTy->isPointerTy() && isStructPtrTy(BaseTy)) {
+      std::vector<Value *> elems =
+          ExpandStruct(gep->getPointerOperand(), F, expend_level - 1);
+      Expanded.insert(Expanded.end(), elems.begin(), elems.end());
```

Comment:
`struct S { int s }; struct T { float t };`
`struct ST { struct s, struct t }` won't be any different from `struct ST { int s, float t }`. Would that be a problem in the future?
You can leave a TODO here so we can track it in the future. (Or fix it now).

Also, can you verify that the struct in a struct is always a pointer? (`BaseTy->isPointerTy() && isStructPtrTy(BaseTy)`)